osquery/tables/system/darwin/keychain_utils.cpp

```diff
@@ -113,6 +113,7 @@ void genAlgorithmProperties(X509* cert,
         }
       }
     }
+    EVP_PKEY_free(pkey);
```

Comment:
Should you check to make sure `pkey != nullptr`? So move this up one line?

Reply:
I realized this while eating dinner. Fixing now

Reply:
Here is the code for EVP_PKEY_free, it  actually does it's own null check so this should still be safe:

```C
void EVP_PKEY_free(EVP_PKEY *x)
{
    int i;

    if (x == NULL)
        return;

    i = CRYPTO_add(&x->references, -1, CRYPTO_LOCK_EVP_PKEY);
#ifdef REF_PRINT
    REF_PRINT("EVP_PKEY", x);
#endif
    if (i > 0)
        return;
#ifdef REF_CHECK
    if (i < 0) {
        fprintf(stderr, "EVP_PKEY_free, bad reference count\n");
        abort();
    }
#endif
    EVP_PKEY_free_it(x);
    if (x->attributes)
        sk_X509_ATTRIBUTE_pop_free(x->attributes, X509_ATTRIBUTE_free);
    OPENSSL_free(x);
}
```